geopandas/array.py

```diff
@@ -567,13 +569,103 @@ def area(self):
         np.ndarray of float
             Area of the geometries.
         """
+        import geopandas
+
+        geodesic_calculation = geopandas.options.geodesic_calculation
+        if geodesic_calculation:
```

Comment:
Not only if `geodesic_calculation` is enabled but also if CRS is set and is geographic.